app-puakaldi-wrapper version 0.1.0

angus-lherrou opened the pull request.

angus-lherrou committed:
fixed typo in KALDI_MEDIA_DIRECTORY: /audio-in -> /audio_in

angus-lherrou committed:
added Dockerfile

angus-lherrou committed:
ffmpeg didn't like absolute path, so move files after encoding

angus-lherrou committed:
subprocess.call doesn't handle '&&' (it isn't a shell, who knew) so split into two commands

angus-lherrou committed:
some style stuff

angus-lherrou committed:
make appmetadata() return str

angus-lherrou committed:
housekeeping

angus-lherrou committed:
wrong dict keys

angus-lherrou committed:
added flag to avoid running kaldi if the last run doesn't need to be repeated

angus-lherrou committed:
Added .dockerignore file

angus-lherrou committed:
Added tests

angus-lherrou committed:
Updated .gitignore

angus-lherrou committed:
Added argument parsing for running the script on a single mmif file instead of starting the Flask server, as well as re-running annotation

angus-lherrou committed:
Updated annotator to use Tokens for alignment

angus-lherrou committed:
Factored out MMIF spec version

angus-lherrou committed:
Fixed pytype complaint

angus-lherrou committed:
added pytype dir to .gitignore

angus-lherrou committed:
Added coverage to .gitignore

angus-lherrou committed:
factored out more directory strings

angus-lherrou committed:
Corrected output for new pattern

angus-lherrou committed:
added comment at top of script

keighrim commented on app.py:
```diff
@@ -1,44 +1,62 @@
 KALDI_OUTPUT_DIR = os.path.join(KALDI_EXPERIMENT_DIR, 'output')
-KALDI_VERSION = 'IDK'
+TRANSCRIPT_DIR = os.path.join(MEDIA_DIRECTORY, 'transcripts')
+KALDI_VERSION = '0.1.0'
```
I'd like to sugest rename this to `APP_VERSION` .

angus-lherrou replied:
Good idea.

keighrim commented on app.py:
```diff
@@ -1,44 +1,62 @@
             "name": "Kaldi Wrapper",
             "description": "This tool wraps the Kaldi ASR tool",
             "vendor": "Team CLAMS",
             "iri": f"http://mmif.clams.ai/apps/kaldi/{KALDI_VERSION}",
```
In addition to `APP_VERSION`, we might also need to keep the record of underlying wrapped tool using something like `WRAPPEE_VERSION` (we don't have a fixed app metadata schema yet). For that there are two versions here, one is the hipstas docker image version (`v1`) and the kaldi version. As kaldi have had a quite complicated history of versioning, the most reliable way to refer a kaldi version would be, I think, the git commit id, which in this image can be obtained via 
```
subprocess.run("git --git-dir=/kaldi/.git rev-parse HEAD".split(), stdout=subprocess.PIPE).stdout.decode('ascii')[:6]
```

I realized that there are several layers of *wrapping* in this app, and how frequent it would the case for other tools. So considering all that, we can concat both versions to use as `WRAPPEE_VERSION` or pick only one.

angus-lherrou replied:
My vote would be just to use the docker image version, since the Kaldi version will only change if the image version changes, right?

keighrim reviewed (commented):
MMIF related code looks good. dockerfile also looks good. Given the ubuntu version issue, we might want to move to base image with a more recent OS, but that's for the future. I left some comments related to appmetadata, which can be easily fixed for this release. We can discuss in more details while we actually add metadata schema.

angus-lherrou committed:
Update .dockerignore

angus-lherrou committed:
Remove locally excluded files from .dockerignore

angus-lherrou committed:
Update .dockerignore

angus-lherrou committed:
Update .dockerignore

Maybe it was a mistake to do this on Github directly but this should be the last one...

angus-lherrou committed:
changed annotate() to return str instead of Mmif, added pretty flag to method parameters (default=False)

angus-lherrou committed:
adjusted argparse to add --pretty flag and to pass along --pretty and --no-kaldi to both the --once operation and the RESTified app

angus-lherrou committed:
changed KALDI_VERSION to APP_VERSION and added WRAPPED_IMAGE global var for "wrappee" field in metadata

angus-lherrou committed:
fixed a missed factoring of the spec version in the tests

angus-lherrou committed:
updated tests; skipping bugged test

angus-lherrou committed:
updated requirements.txt

angus-lherrou committed:
fixed test skip invocation

angus-lherrou commented:
I've made your requested changes and made some additional changes to bring the app up to date with the latest clams-python.

angus-lherrou committed:
correctly parameterized the --once argument (was just running on a fixed path)

keighrim committed:
minor fix on file path

keighrim committed:
now using temp directories for kaldi calls

keighrim reviewed (approved):
code verified and tested.

angus-lherrou committed:
cleanup for code reuse

angus-lherrou committed:
updated TestAnnotation to point to right test output
